### ISSUE TITLE
Alignment of flags to a point when using `onSeries` 

### ISSUE THREAD

**Issue by xPaw** [highcharts/highcharts#21904]
See https://jsfiddle.net/dgLybex6/ for a demo

There is a previous issue for this problem but it was closed as stale: #1528

Specifically I would like way of aligning flags to the closest point on the series, this problem shows up more often when using `dataGrouping`, but can also show when the flag is simply on a `x` value that does not match any point value in the data series.

![image](https://github.com/user-attachments/assets/4f7f4544-4b1d-4793-86a4-4ad409ab4fd6)

Another example: (I would like the flag to be on the peak here, as the data got grouped into months)
![image](https://github.com/user-attachments/assets/6cc6327f-ae62-4ba0-8b8c-dee06fd35ccd)

**Comment by karolkolodziej** [comment]
@xPaw If the flag data is declared like 
```js
    { x: 1360065600000, text: "52 Week High: 41.34", title: "H" },
``` 
How would you reposition it to the nearest point? 
Do you think about something like `snapToClosestPoint` property or do you have some other idea for it?

**Comment by xPaw** [comment]
Snapping is what I had in mind, yes. One might consider forcing it to snap to the left or right instead of closest based on how data works out. For me I would prefer snap to left point, because when data is grouped by month (using high approximation) and a flag lands somewhere late in the month, it's still related to the same month.

**Comment by pawelfus** [comment]
Thank you for sharing the idea! An idea of the implementation can be found below. Let us know what do you think about this solution.

## Plugin

Demo:  https://jsfiddle.net/BlackLabel/bcLft3h5/

API:
- `series.snapTo = 'left' | 'right' | 'closest'` - snap to the left/right/closest point in the underlying series
- `series.onSeries` must be defined
- works when DataGrouping is ON (to disable, search for the` // apply changes only when DataGrouping is ON` comment and remove the condition)

Snippet:
```js
(function (H) {
  const searchers = {
    closest: function (point, onSeries) {
      return onSeries.searchPoint(
        { chartX: point.plotX, chartY: point.plotY, type: "touchstart" }, // build kd-tree now
        true,
      )
    },
    right: function (point, onSeries) {
      return onSeries.points.find((p) => point.x <= p.x)
    },
    left: function (point, onSeries) {
      return onSeries.points[searchers.right(point, onSeries).index - 1]
    },
  }

  H.wrap(H.seriesTypes.flags.prototype, "translate", function (p) {
    p.apply(this, ...arguments)

    const search = searchers[this.options.snapTo]

    if (search && this.options.onSeries) {
      const onSeries = this.chart.get(this.options.onSeries)

      if (!onSeries || !onSeries.currentDataGrouping) return // apply changes only when DataGrouping is ON

      this.points.forEach((p) => {
        const onPoint = search(p, onSeries)

        if (!onPoint) return

        p.plotX = onPoint.plotX
        p.plotY = onPoint.plotY
      })
    }
  })
})(Highcharts);
```

**Comment by pawelfus** [comment]
Yes, it will work without this check. I added this because when DataGrouping kicks in, there's no way to control the flag's position (without the plugin). However, when you present ungrouped data, you have full control over the flag (the x-value) and know the exact date. For convenience, you can remove that check of course.

**Comment by xPaw** [comment]
Will this be merged into highcharts code, or should I just use that snippet?

**Comment by pawelfus** [comment]
For now please use the snippet above.

**Comment by xPaw** [comment]
One issue I've spotted with this is that setting plot coordinates directly doesn't allow for the overlapping mechanisms for flags to kick in, they just draw on top of each other.

**Comment by pawelfus** [comment]
Nice catch! Thank you for letting us know. I have updated the [demo](https://jsfiddle.net/BlackLabel/bcLft3h5/) and snippet above.
